futures-core/src/task/mod.rs

```diff
@@ -25,10 +20,16 @@ pub use self::atomic_waker::AtomicWaker;
 pub struct Context<'a> {
     waker: &'a Waker,
     map: &'a mut LocalMap,
-    executor: Exec<'a>,
+    executor: Option<&'a mut Executor>,
 }
 
 impl<'a> Context<'a> {
+    /// Create a new task context without the ability to `spawn`. Useful for
+    /// `no_std` contexts.
+    pub fn without_spawn(map: &'a mut LocalMap, waker: &'a Waker) -> Context<'a> {
```

Comment:
Can we add a note about how use of this method is explicitly discouraged outside of `no_std`, and how use of it with an `std`-using library may result in panics?